components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ApisApiServiceImpl.java

```diff
@@ -577,6 +613,88 @@ public Response apisApiIdPut(String apiId, APIDTO body, String ifMatch, MessageC
         return null;
     }
 
+    /**
+     * Check whether the token has APIDTO class level Scope annotation
+     * @return true if the token has APIDTO class level Scope annotation
+     */
+    private boolean checkClassScopeAnnotation(org.wso2.carbon.apimgt.rest.api.util.annotations.Scope[] apiDtoClassAnnotatedScopes, String[] tokenScopes) {
+
+        for (org.wso2.carbon.apimgt.rest.api.util.annotations.Scope classAnnotation : apiDtoClassAnnotatedScopes) {
+            for (String tokenScope : tokenScopes) {
+                if (classAnnotation.name().equals(tokenScope)) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Get the API DTO object in which the API field values are overridden with the user passed new values
+     * @throws APIManagementException
+     */
+    private APIDTO getFieldOverriddenAPIDTO(APIDTO apidto, API originalAPI,
+                                            String[] tokenScopes) throws APIManagementException {
+
+        APIDTO originalApiDTO;
+        APIDTO updatedAPIDTO;
+
+        try {
+            originalApiDTO = APIMappingUtil.fromAPItoDTO(originalAPI);
+
+            Field[] fields = APIDTO.class.getDeclaredFields();
+            ObjectMapper mapper = new ObjectMapper();
+            String newApiDtoJsonString = mapper.writeValueAsString(apidto);
+            JSONParser parser = new JSONParser();
+            JSONObject newApiDtoJson = (JSONObject) parser.parse(newApiDtoJsonString);
+
+            String originalApiDtoJsonString = mapper.writeValueAsString(originalApiDTO);
+            JSONObject originalApiDtoJson = (JSONObject) parser.parse(originalApiDtoJsonString);
+
+            for (Field field : fields) {
+                org.wso2.carbon.apimgt.rest.api.util.annotations.Scope[] fieldAnnotatedScopes =
+                        field.getAnnotationsByType(org.wso2.carbon.apimgt.rest.api.util.annotations.Scope.class);
+
+                Gson gsonBuilder = new GsonBuilder().create();
```

Comment:
Not a good idea to use gson. We are currently using Jackson in other places. Those are two different serializer/deserializers and better not to use both at the same time. Apart from that, DTOs are using Jackson annotations. So it is better to use Jackson for this.x

Reply:
Fixed with bac9808